internal/exec/stages/disks/luks.go

```diff
@@ -282,6 +285,21 @@ func (s *stage) createLuks(config types.Config) error {
 				config = string(clevisJson)
 			}
 
+			if len(luks.Clevis.Tang) > 0 {
+				// We cannot guarantee that networking is up yet, fetch the tang
+				// server advertisement to ensure that networking is up before
+				// we pass the device to clevis
+				u, err := url.Parse(luks.Clevis.Tang[0].URL)
+				if err != nil {
+					return fmt.Errorf("parsing tang URL: %v", err)
+				}
+				u.Path = path.Join(u.Path, "adv")
```

Comment:
This is getting into implementation details of how Clevis and Tang work. Is this exactly what Clevis does? (I.e. a `GET` on `$url/adv`?).

Hmm... but actually we don't even strictly need an HTTP 200 here, we just need to check that we reached the server, right? Would it make sense to instead just drop this internal knowledge (which could in theory change) and just try to `GET` (or even just `HEAD`) the Tang URL repeatedly until we get any HTTP response? (Even if it's a 404 or something.)

Reply:
Effectively yes; iirc Clevis itself is actually calling `/adv/{key_id}`. [The Tang protocol docs](https://github.com/latchset/tang#tang-protocol) list it as a public endpoint for fetching public keys so I figured it'd be a safe bet to use that path.

In theory yes. Is your idea to avoid the minor implementation details of the protocol? I felt it was a pretty safe trade-off since it's mentioned in the protocol docs and wouldn't require additional checking into exactly which error we were potentially receiving.

Reply:
Ack thanks. Looks like a well-known and documented API endpoint, so I retract my suggestion.